src/awaitVercelDeployment.ts

```diff
@@ -1,38 +1,50 @@
 import * as core from '@actions/core';
 import fetch from '@adobe/node-fetch-retry';
+import { setTimeout } from 'timers/promises';
 import { VERCEL_BASE_API_ENDPOINT } from './config';
 import { VercelDeployment } from './types/VercelDeployment';
 
+interface Options {
+  /** Base url of the Vercel deployment to await for */
+  url: string;
+  /** Duration (in milliseconds) to wait for a terminal deployment status */
+  timeout: number;
+  /** Duration (in milliseconds) to wait in between polled Vercel API requests */
+  pollInterval: number;
```

Comment:
I pulled this into a single object argument with named parameters, mostly to de-risk an ordering bug with passing in multiple `number` arguments.